app/src/main/java/com/epfl/drawyourpath/mainpage/fragments/helperClasses/ChatAdapter.kt

```diff
@@ -0,0 +1,35 @@
+package com.epfl.drawyourpath.mainpage.fragments.helperClasses
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import android.widget.TextView
+import androidx.recyclerview.widget.RecyclerView
+import com.epfl.drawyourpath.R
+import com.epfl.drawyourpath.chat.Chat
+
+class ChatAdapter(private val chatList: List<Chat>, private val clickListener: (Chat) -> Unit) :
+    RecyclerView.Adapter<ChatAdapter.ChatViewHolder>() {
+
+    inner class ChatViewHolder(view: android.view.View) : RecyclerView.ViewHolder(view) {
+        val chatTitleTextView: TextView = view.findViewById(R.id.chatTitleTextView)
+
+        fun bind(chat: Chat, clickListener: (Chat) -> Unit) {
+            itemView.setOnClickListener { clickListener(chat) }
+        }
+    }
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ChatViewHolder {
+        val view = LayoutInflater.from(parent.context)
+            .inflate(R.layout.item_chat_list, parent, false)
+        return ChatViewHolder(view)
+    }
+
+    override fun onBindViewHolder(holder: ChatViewHolder, position: Int) {
+        val chat = chatList[position]
+        val chatTitle = "Chat ${position + 1}" // Customize this to display the desired chat title

```

Comment:
Yes chats should have custom names, that will be a great addition !
Maybe add a TODO to be more clear ?

Reply:
See the answer to Hugo above